pos_lot_selection/models/pos_session.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2024 Dixmit
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class PosSession(models.Model):
+    _inherit = "pos.session"
+
+    def _loader_params_product_product(self):
+        result = super()._loader_params_product_product()
+        result["search_params"]["fields"].append("available_lot_for_pos_ids")
+        return result
+
+    def get_pos_ui_product_product_by_params(self, custom_search_params):
+        return super(
+            PosSession, self.with_company(self.company_id.id)
```

Comment:
You may have a good reason to update the context  `company`. But i cannot find that reason myself, i see most of the functions have forced by  the company of point of sale, take this as an example:
![image](https://github.com/nguyenminhchien/pos/assets/8724793/7b0cf7f1-4214-49ff-9e38-2402403fccb4)

Reply:
The original function is doing the searc_read without setting the company:

![image](https://github.com/nguyenminhchien/pos/assets/28590170/69aa0d37-dbce-40f1-9fc1-a104a94cf8b2)

It is just a safety guard. I don't get the problem